convert2rhel/checks.py

```diff
@@ -449,3 +453,47 @@ def _bad_kernel_substring(kernel_release):
         )
         return True
     return False
+
+
+def check_package_updates():
+    """Ensures that the system packages installed are up-to-date."""
+    logger.task("Prepare: Checking if the packages are up-to-date")
+
+    yum_stdout, _ = run_subprocess("yum check-update -q", print_output=False)
+
+    # There is a whole blank line before outputting the packages to update
+    # and to output the real counting, we just -1 the counting.
+    packages_to_update_count = yum_stdout.count("\n") - 1
+
```

Comment:
It would be slower but more robust to split the output and filter out the lines we don't want.  I also noticed that when I ran yum check-update -q locally, I got other, non package lines:
> vim-minimal.x86_64                             2:8.2.3568-1.fc33                        updates-testing
> Security: kernel-core-5.14.15-100.fc33.x86_64 is an installed security update
> Security: kernel-core-5.14.13-100.fc33.x86_64 is the currently running version

What do you think about:

``` python
lines = yum_stdout.split("\n")

# Filter lines that only have whitespace or are blank
lines = (line for line in yum_stdout.split("\n") if line.strip())
# Filter informational lines that don't have to do with available packages
lines = (line for line in lines if not line.endswith("is an installed security update"))
packages = [line for line in lines if not line.endswith("is the currently running version")]

packages_to_update_count = len(packages)
```

Reply:
I agree with @SpyTec, the code looks way cleaner and easy to understand with the regex. I tested and got all the packages that needed to update in one go, and if we have to output to the user in the future about the packages, we can just iterate over the list and show the names to them.

Reply:
<nod> Regexes can simplify surrounding code.  However, they mean you have a regex to maintain and understand. So do be mindful of that.

https://www.explainxkcd.com/wiki/index.php/1171:_Perl_Problems

Reply:
The regex would not handle well the situation with the security update @abadger mentioned - it would give us two packages to update even though there's none. So, I'm for the @abadger's approach.

Reply:
@j-mracek, in the output of `yum check-update` shared by @abadger in https://github.com/oamg/convert2rhel/pull/356/files#r741452001 (`is an installed security update`/`is the currently running version`) we can see the same NEVRA at both lines. I suppose the NEVRAs can differ, right? Would in such a case such NEVRA appear with the "Security:" prefix or would it appear on a separate line without the prefix as well?

Also, is there any yum/dnf python API function that would give us the information provided by the `check-update` CLI command?

Reply:
I am little bit lost here, but I will try my best. There are multiple possibilities how to receive information provided by check-update command. Try use Query class with filters updates, and security. Please can you describe exactly what set of packages are you looking for? Then I can look in implementation.

Reply:
@j-mracek We are looking for general packages updates. We are not aiming at a specific set of packages, but rather, if there is any package to be updated in the system. Since @abadger provided us with a different output from the ones I've got in my testing, we were wondering if `yum check-update` could give us any different messages that we are not aware of.

Reply:
What about:
In case that you can directly play with goal in base
base.upgrade_all()
base.resolve()
Then iterate over base.transaction
for x in base.transaction:
    print(x, x.action_name)

Reply:
@j-mracek For some reason, I was not able to manage to get your code snippet working. It doesn't output anything for me when I run it, maybe I'm doing something wrong? 

```python
import dnf

db = dnf.Base()

db.fill_sack()

db.upgrade_all()
db.resolve()

[print(package) for package in db.transaction]
```

I also tried like this: 

```python
import dnf
db = dnf.Base()

db.fill_sack()

query = db.sack.query()
installed = query.upgrades()

print("Installed dnf package:")
for pkg in list(installed):
    print(pkg, pkg.reponame)
```

But none of the above worked... Already checked the output of `dnf check-update` command and it reported I have 50 packages to update.

Also, @j-mracek, I've checked the code for dnf/yum regarding the command `check-update` and both seem to use a different method. Do you think we could also try to do to the same as the tool? Or is the solution you shared with us easier to implement?

Thanks in advance for your help!

Reply:
Do you still need any assistance?

Reply:
@j-mracek, for now, I think I'm all good! Thanks for your helping and patience!